SAML auth with vSphere and zones API not working

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

The vSphere cloud provider is unable to fetch the node failure domain labels when using VCP with SAML auth because of an authentication failure.  For context, the SAML token auth is described here - https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/saml-token-authentication.html.  SAML auth support for the zones api was introduced in this PR - https://github.com/kubernetes/kubernetes/pull/75515, which was cherry-picked to 1.14.1 here - https://github.com/kubernetes/kubernetes/pull/75742

I've tested the credentials I'm using with govc to ensure that I'm able to get a token:
```
$token = govc session.login -l -u vcdo -cert keys\public.crt -key keys\private.key -issue -debug
```
The credentials also work fine when using VCP without the node labels feature. 

Relevant kubelet logs: 
```
I0502 14:31:45.856013    5908 generic.go:189] GenericPLEG: Relisting
I0502 14:31:46.775592    5908 connection.go:142] SessionManager.LoginByToken with certificate
I0502 14:31:46.858589    5908 generic.go:189] GenericPLEG: Relisting
I0502 14:31:46.916706    5908 connection.go:209] New session ID for 'VSPHERE.LOCAL\k8s-vcp-dolap' = 52109852-8b67-9901-c095-6f21d8210c03
I0502 14:31:47.162687    5908 datacenter.go:99] VirtualMachine:vm-82 host is HostSystem:host-65
I0502 14:31:47.833807    5908 config.go:100] Looking for [api], have seen map[api:{}]
I0502 14:31:47.833807    5908 kubelet.go:1972] SyncLoop (housekeeping)
I0502 14:31:47.859584    5908 generic.go:189] GenericPLEG: Relisting
I0502 14:31:48.693024    5908 cni.go:182] Using CNI configuration file C:\k\cni\config\cni.conf
I0502 14:31:48.853838    5908 kubelet.go:2167] Container runtime status: Runtime Conditions: RuntimeReady=true reason: message:, NetworkReady=true reason: message:
I0502 14:31:48.860848    5908 generic.go:189] GenericPLEG: Relisting
I0502 14:31:49.833687    5908 config.go:100] Looking for [api], have seen map[api:{}]
I0502 14:31:49.833687    5908 kubelet.go:1972] SyncLoop (housekeeping)
I0502 14:31:49.863597    5908 generic.go:189] GenericPLEG: Relisting
E0502 14:31:49.992408    5908 vsphere.go:1538] Get zone for node ul-win-node-5k: POST https://vcdo.production.com:443/rest/com/vmware/cis/session: 401 Unauthorized
E0502 14:31:49.992408    5908 kubelet_node_status.go:68] Unable to construct v1.Node object for kubelet: failed to get zone from cloud provider: POST https://vcdo.production.com:443/rest/com/vmware/cis/session: 401 Unauthorized
I0502 14:31:50.194250    5908 kubelet_node_status.go:283] Setting node annotation to enable volume controller attach/detach
I0502 14:31:50.453104    5908 datacenter.go:99] VirtualMachine:vm-82 host is HostSystem:host-65
```

**What you expected to happen**:
VCP should add failure domain labels to the node object.

**How to reproduce it (as minimally and precisely as possible)**:
Setup SAML auth for VCP with zone labeling enabled. 

**Anything else we need to know?**:
There was a previous issue that was raised (https://github.com/kubernetes/kubernetes/issues/75511) but that was before SAML auth support for zones was implemented.  

This could potentially be an issue with govc handling of SAML tokens since even with the Administrator role assigned to the user, I'm still unable to list tags.
```
> $ENV:GOVC_LOGIN_TOKEN = govc session.login -l -u vcdo -cert keys\public.crt -key keys\private.key -issue
> govc tags.attached.ls -json -r /uunet
C:\Users\rhockenbury\Tools\govc.exe: POST https://vcdo/rest/com/vmware/cis/session: 401 Unauthorized
```

The role and keys were generated with the following:
```
 $manageK8sVms= @("Resource.AssignVMToPool","VirtualMachine.Config.AddExistingDisk","VirtualMachine.Config.AddNewDisk","VirtualMachine.Config.AddRemoveDevice","VirtualMachine.Config.RemoveDisk","VirtualMachine.Inventory.Create","VirtualMachine.Inventory.Delete","VirtualMachine.Config.Settings","DataStore.AllocateSpace","DataStore.FileManagement","StorageProfile.View","System.Read","System.Anonymous","System.view")
New-ViRole -name "manage-k8s-devops-lab" -privilege (Get-ViPrivilege -id $manageK8sVms)
set-ViRole -role "manage-k8s-devops-lab" -AddPrivilege (Get-VIPrivilege -id $manageK8sVms)

openssl req -newkey rsa:2048 -x509 -days 365 -nodes -keyout k8s-vcp-dolab.key -out k8s-vcp-dolab.crt -subj "/C=US/ST=CA/L=SF/O=VMware/OU=CNA/CN=www.vmware.com"

/usr/lib/vmware-vmafd/bin/dir-cli service create --name k8s-vcp-dolap --cert k8s-vcp-dolab.crt 
```

**Environment**:
- Kubernetes version (use `kubectl version`):
1.14.1
- Cloud provider or hardware configuration:
vSphere
- OS (e.g: `cat /etc/os-release`):
Linux & Windows

Seeing a lot of this exception:

```
2019-06-07T17:52:41.459Z | ERROR | sso5                      | SoapBindingImpl                | SOAP fault
com.sun.xml.internal.ws.fault.ServerSOAPFaultException: Client received SOAP Fault from server: Cannot continue delegation chain Please see the server log to find more detail regarding exact cause of the failure.
	at com.sun.xml.internal.ws.fault.SOAP11Fault.getProtocolException(SOAP11Fault.java:178)
	at com.sun.xml.internal.ws.fault.SOAPFaultBuilder.createException(SOAPFaultBuilder.java:116)
	at com.sun.xml.internal.ws.client.dispatch.DispatchImpl.doInvoke(DispatchImpl.java:259)
	at com.sun.xml.internal.ws.client.dispatch.DispatchImpl.invoke(DispatchImpl.java:289)
	at com.vmware.vim.sso.client.impl.SoapBindingImpl.sendMessage(SoapBindingImpl.java:161)
	at com.vmware.vim.sso.client.impl.SoapBindingImpl.sendMessage(SoapBindingImpl.java:114)
	at com.vmware.vim.sso.client.impl.SecurityTokenServiceImpl$RequestResponseProcessor.sendRequest(SecurityTokenServiceImpl.java:927)
	at com.vmware.vim.sso.client.impl.SecurityTokenServiceImpl$RequestResponseProcessor.executeRoundtrip(SecurityTokenServiceImpl.java:856)
	at com.vmware.vim.sso.client.impl.SecurityTokenServiceImpl.acquireTokenByCertificate(SecurityTokenServiceImpl.java:477)
	at com.vmware.vapi.endpoint.sso.context.StsFacadeImpl.acquireActAsToken(StsFacadeImpl.java:37)
	at com.vmware.vapi.endpoint.sso.context.SecurityContextUtil.acquireAndRenewActAsToken(SecurityContextUtil.java:303)
	at com.vmware.vapi.endpoint.sso.context.SecurityContextUtil.access$000(SecurityContextUtil.java:38)
	at com.vmware.vapi.endpoint.sso.context.SecurityContextUtil$2.run(SecurityContextUtil.java:170)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
2019-06-07T17:52:41.459Z | WARN  | sso5                      | SecurityContextUtil            | Failed to acquire Act As token.
com.vmware.vapi.endpoint.sso.StsException: com.vmware.vim.sso.client.exception.InvalidTokenRequestException: Request is invalid: ns0:InvalidRequest: Cannot continue delegation chain
	at com.vmware.vapi.endpoint.sso.context.StsFacadeImpl.acquireActAsToken(StsFacadeImpl.java:73)
	at com.vmware.vapi.endpoint.sso.context.SecurityContextUtil.acquireAndRenewActAsToken(SecurityContextUtil.java:303)
	at com.vmware.vapi.endpoint.sso.context.SecurityContextUtil.access$000(SecurityContextUtil.java:38)
	at com.vmware.vapi.endpoint.sso.context.SecurityContextUtil$2.run(SecurityContextUtil.java:170)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.vmware.vim.sso.client.exception.InvalidTokenRequestException: Request is invalid: ns0:InvalidRequest: Cannot continue delegation chain
	at com.vmware.vim.sso.client.impl.SecurityTokenServiceImpl$RequestResponseProcessor.handleFaultCondition(SecurityTokenServiceImpl.java:1077)
	at com.vmware.vim.sso.client.impl.SecurityTokenServiceImpl$RequestResponseProcessor.sendRequest(SecurityTokenServiceImpl.java:932)
	at com.vmware.vim.sso.client.impl.SecurityTokenServiceImpl$RequestResponseProcessor.executeRoundtrip(SecurityTokenServiceImpl.java:856)
	at com.vmware.vim.sso.client.impl.SecurityTokenServiceImpl.acquireTokenByCertificate(SecurityTokenServiceImpl.java:477)
	at com.vmware.vapi.endpoint.sso.context.StsFacadeImpl.acquireActAsToken(StsFacadeImpl.java:37)
	... 6 more
2019-06-07T17:52:41.459Z | INFO  | sso5                      | BaseSessionImpl                | Cannot authenticate user login request com.vmware.vapi.cis.authn.SecurityContextFactory$1@1239e696 with error com.vmware.vapi.endpoint.sso.StsException: com.vmware.vim.sso.client.exception.InvalidTokenRequestException: Request is invalid: ns0:InvalidRequest: Cannot continue delegation chain.
```

--- a/staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/connection.go
+++ b/staging/src/k8s.io/legacy-cloud-providers/vsphere/vclib/connection.go
@@ -111,6 +111,7 @@ func (connection *VSphereConnection) Signer(ctx context.Context, client *vim25.C
 
 	req := sts.TokenRequest{
 		Certificate: &cert,
+		Delegatable: true,
 	}
 
 	signer, err := tokens.Issue(ctx, req)

